File: app/src/main/java/com/main/climbingdiary/database/entities/AreaRepository.java
package com.main.climbingdiary.database.entities;

import android.database.Cursor;

import com.main.climbingdiary.database.TaskRepository;
import com.main.climbingdiary.models.SportType;

import org.chalup.microorm.MicroOrm;

import java.util.ArrayList;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class AreaRepository {

    private final TaskRepository taskRepository;

    private static AreaRepository INSTANZ = null;
    private final MicroOrm uOrm = new MicroOrm();

    public static synchronized AreaRepository getInstance(){
        if(INSTANZ==null){
            INSTANZ = new AreaRepository(TaskRepository.getInstance());
        }
        return INSTANZ;
    }

    public Area getAreaByAreaNameAndSectorName(String sectorName, String areaName){
        Cursor cursor = taskRepository.getAreaIdByAreaNameAndSectorName(sectorName, areaName);
        return uOrm.fromCursor(cursor, Area.class);
    }

    public ArrayList<Area> getAreaList(){
        ArrayList<Area> _area_list = new ArrayList<>();
        Cursor cursor = taskRepository.getAllAreas();
        if (cursor != null) {
            while (cursor.moveToNext()) {
                _area_list.add(uOrm.fromCursor(cursor, Area.class));
            }
        }
        return _area_list;
    }
    public ArrayList<String> getAreaNameList(){
        ArrayList<String> _area_list = new ArrayList<>();
        Cursor cursor = taskRepository.getAllAreas();
        if (cursor != null) {
            while (!cursor.isAfterLast()) {
                String name = cursor.getString(1);
                _area_list.add(name);
                cursor.moveToNext();
            }
        }
        return _area_list;
    }
}

File: app/src/main/java/com/main/climbingdiary/database/entities/AreaRepository.kt
package com.main.climbingdiary.database.entities

import android.database.Cursor
import android.database.CursorIndexOutOfBoundsException
import com.main.climbingdiary.database.TaskRepository
import org.chalup.microorm.MicroOrm

object AreaRepository {

    private val uOrm = MicroOrm()

    fun getAreaByAreaNameAndSectorName(sectorName: String?, areaName: String?): Area? {
        try {
            val cursor: Cursor =
                TaskRepository.getAreaIdByAreaNameAndSectorName(sectorName, areaName)
            return uOrm.fromCursor(cursor, Area::class.java)
        } catch (ex: CursorIndexOutOfBoundsException) {
            return null
        }
    }

    fun insertArea(area: Area): Boolean {
        return TaskRepository.insertArea(area)
    }

    fun getAreaByName(name: String): Area? {
        //Fixme: return null if not exists
        try {
            val cursor: Cursor = TaskRepository.getAreaByAreaName(name)
            return uOrm.fromCursor(cursor, Area::class.java)
        } catch (ex: CursorIndexOutOfBoundsException) {
            return null
        }
    }

    fun getAreaList(): ArrayList<Area> {
        val area_list = ArrayList<Area>()
        val cursor: Cursor = TaskRepository.getAllAreas()
        while (cursor.moveToNext()) {
            area_list.add(
                uOrm.fromCursor(
                    cursor,
                    Area::class.java
                )
            )
        }
        return area_list
    }

    fun getAreaNameList(): ArrayList<String> {
        val area_list = ArrayList<String>()
        val cursor: Cursor = TaskRepository.getAllAreas()
        while (!cursor.isAfterLast) {
            val name = cursor.getString(1)
            area_list.add(name)
            cursor.moveToNext()
        }
        return area_list
    }
}